### DISCUSSION TITLE
Conditionally setting #[pymethods] attributes

### DISCUSSION THREAD

**Discussion by ju6ge** [PyO3/pyo3#3373] [Questions]
Hi guys,

I am trying to write rust code and generate python bindings as well as r bindings using `extendr`. Right now I am hitting an issue that leads me having to duplicate code. Here is an example:

```rust
#[cfg_attr(feature="py-bindings", pyclass)]
#[cfg_attr(feature="r-bindings", extendr)]
pub struct TestClass {
     /*some fileds */
}

#[cfg(feature="py-bindings")
#[pymethods]
impl TestClass {
    #[staticmethod]
    pub fn static_function() -> Self { /* ... */ }
}


#[cfg(feature="r-bindings")
#[extendr]
impl TestClass {
    pub fn static_function() -> Self { /* ... */ }
}
```

Sadly I can not do the following: 
```rust
#[cfg_attr(feature="py-bindings", pymethods)]
#[cfg_attr(feature="r-bindings", extendr)]
impl TestClass {
    #[cfg_attr(feature="py-bindings", staticmethod)]
    pub fn static_function() -> Self { /* ... */ }
}
```
I get an error that `staticmethod` is not known within this scope. When trying to build wtih the `py-bindings` feature enabled. Building with the `r-bindings` feature enabled works.

The implementation stays the same, yet I have to write the code twice. This is not very nice.

Is there a way to conditionally add the attributes?

Kind regards,
ju6ge

**Comment by adamreichold** [comment]
I think this might be the functionality discussed in https://github.com/PyO3/pyo3/pull/2786, is it not?
